test/e2e/E2ETest.t.sol

```diff
@@ -135,7 +137,32 @@ abstract contract E2ETest is BaseUser, Base_CType, Common {
     (uint256 _expectedCollateral,) = collateralAuctionHouse[_cType()].getCollateralBought(1, _amountToBid);
     assertEq(_expectedCollateral, COLLATERAL_AMOUNT);
 
-    _joinCoins(address(this), _amountToBid);
+    _buyCollateral(address(this), address(collateralAuctionHouse[_cType()]), 1, _expectedCollateral, _amountToBid);
+
+    uint256 _decimals = collateral[_cType()].decimals();
+    uint256 _collateralWei = _expectedCollateral / 10 ** (18 - _decimals);
+    assertEq(collateral[_cType()].balanceOf(address(this)) - _initialBalance, _collateralWei);
+
+    // NOTE: auctions(1) is deleted
+    ICollateralAuctionHouse.Auction memory _auction = collateralAuctionHouse[_cType()].auctions(1);
+    assertEq(_auction.amountToSell, 0);
+    assertEq(_auction.amountToRaise, 0);
+  }
+
+  function test_collateral_auction_full_raise() public {
+    _generateDebt(address(this), address(collateralJoin[_cType()]), int256(COLLATERAL_AMOUNT), int256(DEBT_AMOUNT));
+    uint256 _initialBalance = collateral[_cType()].balanceOf(address(this));
+    _setCollateralPrice(_cType(), PRICE_DROP);
+    _liquidateSAFE(_cType(), address(this));
+    _setCollateralPrice(_cType(), INITIAL_PRICE);
+
+    uint256 _amountToBid = Math.wmul(DEBT_AMOUNT, LIQUIDATION_PENALTY);
+    // NOTE: getExpectedCollateralBought doesn't have a previous reference (lastReadRedemptionPrice)
+    (uint256 _expectedCollateral,) = collateralAuctionHouse[_cType()].getCollateralBought(1, _amountToBid);
+    assertLt(_expectedCollateral, COLLATERAL_AMOUNT);
+
+    _generateDebt(address(this), address(collateralJoin[_cType()]), int256(COLLATERAL_AMOUNT), int256(DEBT_AMOUNT));
+    _generateDebt(address(this), address(collateralJoin[_cType()]), int256(COLLATERAL_AMOUNT), int256(DEBT_AMOUNT));
```

Comment:
`_generateDebt()` is being called twice so as to avoid a `ERC20InsufficientBalance` reversion. Should it be called once?

Reply:
how can you not generate double collateral? not following